### PR TITLE
Add compilers metapackage

### PR DESCRIPTION
**Pull request by chrisburr**
@duncanmmacleod @scopatz This builds on #7739 to remove the assumption of which compiler is used on each platform when testing the symlinks. It also avoids the linter bug.

@duncanmmacleod I've invited you to my fork in case you want to make any changes.

### CODE REVIEW

File: recipes/compilers/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,217 @@
...
+
+package:
+  name: compilers
+  version: 1.0.0
+
+build:
+  number: 0
+
+outputs:
+  - name: binutils
```
Comment (chrisburr) [inline recipes/compilers/meta.yaml]: This should probably be called something different. Any ideas?

File: recipes/compilers/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,217 @@
...
+    requirements:
+      host:
+        - binutils_linux-64  # [linux]
+        - cctools  # [osx]
+        - ld64  # [osx]
+      run:
+        - binutils_linux-64  # [linux]
+        - cctools  # [osx]
+        - ld64  # [osx]
+    script: build-symlinks.sh
```
Comment (duncanmmacleod) [inline recipes/compilers/meta.yaml]: I think this needs a `# [unix]` selector.

Comment (chrisburr) [reply recipes/compilers/meta.yaml]: Yeah that fixed it, for some reason Appveyor didn't show up in the checks for the old PR.

File: recipes/compilers/meta.yaml (hunk 1 above)
Comment (duncanmmacleod) [reply recipes/compilers/meta.yaml]: I think on linux this is normally called `binutils`, on macports this is called [`cctools`](https://github.com/macports/macports-ports/blob/master/devel/cctools/Portfile). I think `binutils` is fine.

**Comment by duncanmmacleod** [comment]
@chrisburr, do you think this is ready to go now?

**Comment by chrisburr** [comment]
Yup, LGTM 👍

**Comment by duncanmmacleod** [comment]
@conda-forge/core, I think this is ready for review, thanks.

**Comment by scopatz** [comment]
LGTM thanks!

**Pull request merged**